Gerrit testing

CheyenneWills opened the pull request:
Changes to make it easier to test against a different gerrit system.

CheyenneWills committed:
Use ansible to set notification list

Move the list of email addresses out of master.cfg and set via
ansible variables.

CheyenneWills committed:
Ensure known_hosts contains gerrit host key

When testing, the host key for the gerrit system may change.

Use ssh-keyscan to dynamically obtain the host key.

CheyenneWills committed:
Disable git ssl checking when testing

If the remote repo is accessed via https and the remote site is using a
self-signed key, the worker(s) will fail with a certificate error.

Disable the git ssl verification when testing.

CheyenneWills commented on files/master.cfg:
```diff
@@ -502,6 +505,18 @@ add_parked_builder('fedora29-x86_64', env=env_fedora, factory=Make(checking=True
 
+{% else %}
+{%     if afsbotcfg_testbuilders is not defined %}
+add_gerrit_dev('centos73-x86_64')
```
One possibility as an alternative here would be to maybe use the templating include facility
{% include './templates/testworkers.j2' %}

Or something along those lines.

meffie replied:
I think we should move the data to the inventory and use jinja2 to create the the configuration.

Originally, we use python as a macro language to set the BuildmasterConfig dictionary.  But now we want to have different BuildmasterConfig dictionaries depending on the inventory host variables.

It might be better to split this large pull request into several so can migrate.

meffie commented on files/master.cfg:
```diff
@@ -31,12 +30,22 @@ class WorkerAdmins:
+{% endraw %}
+# Configurations controlled by ansible
+
+cfg_title = '{{ afsbotcfg_title }}'                 # Title for buildbot banner
```
If you use a class or attribute dictionary, you don't have to change the rest of this file, and it would look much nicer and less like c-language in python.

CheyenneWills replied:
ack.. I'll just use a settings = { ... }

meffie commented on files/master.cfg:
```diff
@@ -31,12 +30,22 @@ class WorkerAdmins:
             return None
 
+{% endraw %}
+# Configurations controlled by ansible
```
This comment is unnecessary. This is a jinja2 template so it's just redundant that it is rendered.

CheyenneWills replied:
ack

CheyenneWills committed:
Set WEB UI strings via ansible

The title for the top banner on the web page and the url link are
hardcoded strings.  In addition the settings.title is really being used
as the string for in gerrit summary message.

Set the WEB UI via ansible variables.
Add a new settings that contains the gerrit summary message title.

settings.title is the string that is displayed at the top of the
buildbot web page (and is limited in length by buildbot).

settings.summary_title is the string that is displayed in the build
summary message sent back to gerrit.

CheyenneWills committed:
Remove settings.py, set cfg via ansible

Update master.cfg to just use ansible to define the settings instead
of importing settings.py

Remove the settings.py file

meffie reviewed (approved):
+1

meffie commented on files/master.cfg:
```diff
@@ -31,6 +30,18 @@ class WorkerAdmins:
+settings.repo = '{{ afsbotcfg_repo }}'                   # Gerrit repo
+settings.buildbotURL = '{{ afsbotcfg_url }}'             # URL to buildbot
+settings.www_port = {{ afsbotcfg_www_port }}             # Buildbot port
+settings.build_delay = {{ afsbotcfg_build_delay }}       # Time to delay start of builds
```
This looks good, thanks. I think this would work too:

    class Settings:
         title = '{{ .... }}'
         repo = '{{ ... }}'
         ...

    settings = Settings()

CheyenneWills replied:
Yes -- I just used the pattern from the python doc on namespaces..

meffie commented on .gitignore:
```diff
@@ -1,4 +1,5 @@
 .*
 !.yamllint
 !.ansible-lint
+test/
```
Did we mean to add this to the .gitignore?

CheyenneWills replied:
that was where I was dumping my local env file, ssh keys for a test gerrit  and passwords.ini file.

meffie reviewed (commented):
Looks good. Thanks!

meffie reviewed (commented):
looks good

meffie reviewed (commented):
I guess I dont see yet why we want to do this?

CheyenneWills committed:
Use ansible to set the gerrit connection info

Add documentation for testing
